src/sgame/sg_bot_util.cpp

```diff
@@ -316,26 +316,24 @@ int BotValueOfUpgrades( gentity_t *self )
 // Allow human bots to decide what to buy
 // pre-condition:
 // * weapon is a valid pointer
-// * "upgrades" is an array of at least maxUpgrades elements
+// * "upgrades" is an array of at least upgradesSize elements
 // post-conditions:
 // * "upgrades" contains a list of upgrades to use
 // * "weapon" contains weapon to use
 // * Returns number of upgrades to buy (does not includes weapons)
-int BotGetDesiredBuy( gentity_t *self, weapon_t *weapon, upgrade_t *upgrades, int maxUpgrades )
+int BotGetDesiredBuy( gentity_t *self, weapon_t &weapon, upgrade_t upgrades[], size_t upgradesSize )
 {
-	ASSERT( self && weapon && upgrades );
+	ASSERT( self && upgrades );
 	ASSERT( self->client->pers.team == TEAM_HUMANS ); // only humans can buy
-	ASSERT( maxUpgrades >= 2 ); // because human bots will want up to 2 upgrades...
+	ASSERT( upgradesSize >= 2 ); // because human bots will want up to 2 upgrades...
```

Comment:
I don't understand the comment here, "up to 2"?

Reply:
armor and radar. Other upgrades can not, in current state, be used by bots.